lib/compat/wordpress-6.1/merge-theme-json.php

```diff
@@ -0,0 +1,19 @@
+<?php
+
+function gutenberg_merge_theme_json( $existing_data, $incoming_data, $origin = null ) {
+	if ( $origin === null && doing_filter( 'global_styles_filter_theme' ) ) {
```

Comment:
By filling the `$origin` depending on which filter is running, we avoid asking consumers to do it themselves. Though if the consumer does provide it, we use it instead.

Reply:
I like this. It abstracts away this requirement which might not be easily understood by consumers implementing the filter.